python/vineyard/contrib/ml/pytorch.py

```diff
@@ -130,19 +152,39 @@ def torch_table_resolver(obj, resolver):
     batches = []
     for idx in range(int(meta['__batches_-size'])):
         batches.append(resolver.run(obj.member('__batches_-%d' % idx)))
-    arrow = pa.Table.from_batches(batches).to_pandas()
-    target = torch.tensor(arrow['target'].values)
-    ds = torch.tensor(arrow.drop('target', axis=1).values)
-    return torch.utils.data.TensorDataset(ds, target)
+    return ConcatDataset(batches)
+
+
+def torch_global_tensor_resolver(obj, resolver, **kw):
+    meta = obj.meta
+    num = from_json(meta['num'])
+    partition_index = from_json(meta['partition_index_'])
+    data = []
+    for i in range(partition_index):
+        if meta[f'partition_{i}'].islocal:
+            data.append(resolver.run(obj.member(f'partition_{i}')))
+    return ConcatDataset(data)
+
+
+def torch_global_dataframe_resolver(obj, resolver, **kw):
+    meta = obj.meta
+    num = from_json(meta['num'])
+    partition_index = from_json(meta['partition_index_'])
```

Comment:
this should be num = int(meta['partitions_-size']), here the keyword 'partitions_-size' is from the global dataframe definition in our C++ code. Only matching these keywords between builder and resolver is not enough. You can refer the dask code for the right use of the keywords.